net-vpn/i2pd/patches/i2pd-2.49.0.patchset

```diff
@@ -0,0 +1,88 @@
+From 281170900884ec6307b2d8b84a4bc9695de25bb6 Mon Sep 17 00:00:00 2001
+From: hfsfox <darkkitsunezx128k2309@gmail.com>
+Date: Mon, 18 Sep 2023 13:20:10 +0000
+Subject: fix build and config search paths
+
+
+diff --git a/build/CMakeLists.txt b/build/CMakeLists.txt
+index 3185ffa..bfd1b20 100644
+--- a/build/CMakeLists.txt
++++ b/build/CMakeLists.txt
+@@ -139,6 +139,10 @@ if(APPLE)
+   add_definitions(-DMAC_OSX)
+ endif()
+ 
++if(HAIKU)
++  add_definitions(-D_DEFAULT_SOURCE)
++endif()
++
+ if(MSVC)
+   add_definitions(-DWINVER=0x0600)
+   add_definitions(-D_WIN32_WINNT=0x0600)
+diff --git a/libi2pd/FS.cpp b/libi2pd/FS.cpp
+index 7334550..d38bcc2 100644
+--- a/libi2pd/FS.cpp
++++ b/libi2pd/FS.cpp
+@@ -136,6 +136,14 @@ namespace fs {
+ 		dataDir = (home != NULL && strlen(home) > 0) ? home : "";
+ 		dataDir += "/Library/Application Support/" + appName;
+ 		return;
++#elif defined(__HAIKU__)
++		char *home = getenv("HOME");
++		if (home != NULL && strlen(home) > 0) {
++			dataDir = std::string(home) + "/config/settings/" + appName;
++		} else {
++			dataDir = "/tmp/" + appName;
++		}
++		return;
+ #else /* other unix */
+ #if defined(ANDROID)
+ 		const char * ext = getenv("EXTERNAL_STORAGE");
+diff --git a/libi2pd/I2PEndian.h b/libi2pd/I2PEndian.h
+index 06abf29..681a499 100644
+--- a/libi2pd/I2PEndian.h
++++ b/libi2pd/I2PEndian.h
+@@ -14,7 +14,7 @@
+ #if defined(__FreeBSD__) || defined(__NetBSD__)
+ #include <sys/endian.h>
+ 
+-#elif defined(__linux__) || defined(__FreeBSD_kernel__) || defined(__OpenBSD__) || defined(__GLIBC__)
++#elif defined(__linux__) || defined(__FreeBSD_kernel__) || defined(__OpenBSD__) || defined(__GLIBC__) || defined(__HAIKU__)
+ #include <endian.h>
+ 
+ #elif defined(__APPLE__) && defined(__MACH__)
+diff --git a/libi2pd/util.cpp b/libi2pd/util.cpp
+index 088a7af..47b87b5 100644
+--- a/libi2pd/util.cpp
++++ b/libi2pd/util.cpp
+@@ -27,6 +27,16 @@
+ # include <AvailabilityMacros.h>
+ #endif
+ 
++#if defined(__HAIKU__)
++#include <posix/pthread.h>
++#include <posix/sys/sockio.h>
++#include <posix/sys/ioctl.h>
++#ifndef _DEFAULT_SOURCE
++#define _DEFAULT_SOURCE
++#include <bsd/ifaddrs.h>
++#endif
++#endif
++
+ #ifdef _WIN32
+ #include <stdlib.h>
+ #include <string.h>
+@@ -190,8 +200,10 @@ namespace util
+ 		}
+ 		#pragma warning(pop)
+ 	#else
++	#if !defined(__HAIKU__)
+ 		pthread_setname_np(pthread_self(), name);
```

Comment:
pthread_setname_np should be available on Haiku (see https://github.com/haiku/haiku/commit/44cceee67e056d8e34cb017d5f5c07b9fac874c0)